tools/vegan_diversity/vegan_diversity.xml

```diff
@@ -0,0 +1,88 @@
+        <param name="species_column" label="Group name column" type="data_column" data_ref="input_abundance" value="6" help="Species, phylum, etc"/>
+        <param name="sample_columns" label="Sample count columns" type="data_column" multiple="True" value="2" data_ref="input_abundance" help="Select each column that contains counts"/>
+        <param name="header" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="False" label="Input has a header line"/>
+        <conditional name="index_type">
+            <param name="index_type_selector" type="select" label="Diversity index to use">
+                <option value="shannon" selected="True">shannon</option>
+                <option value="simpson">simpson</option>
+                <option value="invsimpson">invsimpson</option>
+            </param>
+            <when value="shannon">
+                <param name="margin" type="integer" value="1" label="Margin for which the index is computed"/>
+                <param name="base" type="select" label="The logarithm base used in shannon">
+                    <option value="exp(1)" selected="True">exp(1)</option>
+                    <option value="2" selected="False">2</option>
+                </param>
+            </when>
+            <when value="simpson">
+                <param name="margin" type="integer" value="1" label="Margin for which the index is computed"/>
+                <param name="base" type="select" label="The logarithm base used in shannon">
+                    <option value="exp(1)" selected="True">exp(1)</option>
+                    <option value="2" selected="False">2</option>
+                </param>
+            </when>
+            <when value="invsimpson">
+                <param name="margin" type="integer" value="1" label="Margin for which the index is computed"/>
+                <param name="base" type="select" label="The logarithm base used in shannon">
+                    <option value="exp(1)" selected="True">exp(1)</option>
+                    <option value="2" selected="False">2</option>
+                </param>
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data format="tabular" name="output_diversity_index" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_abundance" ftype="tabular" value="vegan_in.tabular"/>
+            <param name="species_column" value="6"/>
+            <param name="sample_columns" value="2"/>
+            <param name="index_type_selector" value="shannon"/>
+            <param name="margin" value="1"/>
+            <param name="base" value="exp(1)"/>
+            <output name="output_slope" ftype="tabular" file="output_diversity_index.tabular" />
+        </test>
+    </tests>
+    <help>
+        <![CDATA[
+Calculate Diversity index using vegan and selected method.
```

Comment:
More help, with hyperlinks and more stuff from https://cran.r-project.org/web/packages/vegan/vignettes/diversity-vegan.pdf + a link to the vignette? Can we ship the vignettes like an image with the tool?